fintech-examples/fintech-ui/src/app/bank/list-accounts/list-accounts.component.ts

```diff
@@ -35,35 +36,55 @@ export class ListAccountsComponent implements OnInit {
     const settings = this.storageService.getSettings();
     const online = !this.storageService.isAfterRedirect() && !settings.cacheLoa;
     this.storageService.setAfterRedirect(false);
-    this.aisService.getAccounts(this.bankId, settings.loa, settings.withBalance, online, settings.consentRequiresAuthentication).subscribe((response) => {
-      switch (response.status) {
-        case 202:
-          this.storageService.setRedirect(
-            response.headers.get(HeaderConfig.HEADER_FIELD_REDIRECT_CODE),
-            response.headers.get(HeaderConfig.HEADER_FIELD_AUTH_ID),
-            response.headers.get(HeaderConfig.HEADER_FIELD_X_XSRF_TOKEN),
-            parseInt(response.headers.get(HeaderConfig.HEADER_FIELD_REDIRECT_X_MAX_AGE), 0),
-            RedirectType.AIS
-          );
-          const r = new RedirectStruct();
-          r.redirectUrl = encodeURIComponent(response.headers.get(HeaderConfig.HEADER_FIELD_LOCATION));
-          r.redirectCode = response.headers.get(HeaderConfig.HEADER_FIELD_REDIRECT_CODE);
-          r.bankId = this.bankId;
-          r.bankName = this.storageService.getBankName();
-          this.router.navigate(['redirect', JSON.stringify(r)], { relativeTo: this.route });
-          break;
-        case 200:
-          // this is added to register url where to forward
-          // if LoT is cancelled after redirect page is displayed
-          // to be removed when issue https://github.com/adorsys/open-banking-gateway/issues/848 is resolved
-          // or Fintech UI refactored
-          this.accounts = response.body.accounts;
-          const loa = [];
-          for (const accountDetail of this.accounts) {
-            loa.push(new AccountStruct(accountDetail.resourceId, accountDetail.iban, accountDetail.name));
-          }
-          this.storageService.setLoa(this.bankId, loa);
-      }
-    });
+
+    const consent: AisConsentRequest = {
+      access: {
+        allPsd2: AllPsd2Enum.ACCOUNTSWITHBALANCES
+      },
+      frequencyPerDay: settings.frequencyPerDay,
+      validUntil: settings.validUntil,
+      recurringIndicator: settings.recurringIndicator,
+      combinedServiceIndicator: settings.combinedServiceIndicator
+    };
+
+    this.aisService
+      .getAccounts(
+        this.bankId,
+        settings.loa,
+        JSON.stringify(consent),
```

Comment:
The consent should be completely optional, so you would set it on settings screen only if needed. Something like StorageService.getConsentOrNull